lib/chef/resource/freebsd_package.rb

```diff
@@ -49,13 +49,11 @@ def ships_with_pkgng?
       end
 
       def assign_provider
-        @provider = if source.to_s =~ /^ports$/i
-                      Chef::Provider::Package::Freebsd::Port
-                    elsif supports_pkgng?
-                      Chef::Provider::Package::Freebsd::Pkgng
-                    else
-                      Chef::Provider::Package::Freebsd::Pkg
-                    end
+        @provider ||= if supports_pkgng?
+                        Chef::Provider::Package::Freebsd::Pkgng
+                      else
+                        Chef::Provider::Package::Freebsd::Pkg
+                      end
```

Comment:
I don't know that provides syntax, sorry. I also asked to give some example or link to documentation, if you want me to rewrite something.

I removed Port because it was failing with multiple packages, an exception is being raised that you cannot specify source with multiple packages. If you suggest to drop multiple support from this pr, this patch becomes nonsense and I will revert it. The idea was to do:
<pre>package 'bash', 'zsh' do
  provider Chef::Provider::Package::Freebsd::Port
end</pre>
instead of specifying `source`.